Stable sessions docs (#11352)
withastro/docs: src/content/docs/en/reference/api-reference.mdx

Allows customizing how the cookie is serialized.

### `session`

<p>

**Type:** `AstroSession`

> <Since v="5.1.0" />

Reviewer: ```suggestion
<Since v="5.7.0" />
```
We should put the first *stable* version here. And, as for the items following, I *would* tend to also mention 5.7.0 *if* you think it's likely more will be added here. If it's not super likely that this will regularly update, then fine to just keep it to the parent.

</p>

> `session` is an object that allows data to be stored between requests for [routes rendered on demand](/en/guides/on-demand-rendering/). It is associated with a cookie that contains the session ID, but differs from [`cookies`](#cookies) because the data itself is not stored in the cookie. This means there are no limits on the amount of data that can be stored in the session, and it can be trusted with secure data. The session is created when first used, and the session cookie is automatically set. The `session` object is `undefined` if no session storage has been configured, or if the current route is prerendered, and will log an error if you try to use it.

Reviewer: ```suggestion
`session` is an object that allows data to be stored between requests for [routes rendered on demand](/en/guides/on-demand-rendering/). 

It is associated with a cookie that contains the session ID only: the data itself is not stored in the cookie. This provides some key advantages over using [`cookies`](#cookies) for data storage: there are no limits on the amount of data that can be stored in the session, and it can be trusted with secure data.

The session is created when first used, and the session cookie is automatically set. The `session` object is `undefined` if no session storage has been configured, or if the current route is prerendered, and will log an error if you try to use it.
```

      const cart = await session.get('cart');
      const newItem = await request.json();
      cart.push(newItem);
      // Save the updated cart to the session
      session.set('cart', cart);
      return Response.json({ cart });
    }
    ```

  </TabItem>
</Tabs>

#### `regenerate()`

<p>

**Type**: `() => void`

</p>

Regenerates the session ID. Best practice is to call this when a user logs in or escalates their privileges, to prevent session fixation attacks.

<Tabs>
  <TabItem label="Astro.session">
    ```astro title="src/pages/welcome.astro" "Astro.session.regenerate()"
    ---
    Astro.session.regenerate();
    ---
    ```
  </TabItem>
  <TabItem label="context.session">
    ```ts title="src/pages/api/login.ts" "session.regenerate()"
    import type { APIContext } from 'astro';

    export async function POST({ session }: APIContext) {
      // Authenticate the user...
      doLogin();
      // Regenerate the session ID to prevent session fixation attacks
      session.regenerate();
      return Response.json({ success: true });
    }
    ```

  </TabItem>
</Tabs>

#### `destroy()`

<p>

**Type**: `() => void`

</p>

> Destroys the session, deleting the cookie and the object from the backend. This should be called when a user logs out or their session is otherwise invalidated.

Reviewer: ```suggestion
Destroys the session, deleting the cookie and the object from the backend. Call this when a user logs out or their session is otherwise invalidated.
```

<Tabs>
  <TabItem label="Astro.session">
    ```astro title="src/pages/logout.astro" "Astro.session.destroy()"
    ---
    Astro.session.destroy();
    return Astro.redirect('/login');
    ---
    ```
  </TabItem>
  <TabItem label="context.session">
    ```ts title="src/pages/api/logout.ts" "session.destroy()"
    import type { APIContext } from 'astro';

    export async function POST({ session }: APIContext) {
      session.destroy();
      return Response.json({ success: true });
    }
    ```

  </TabItem>
</Tabs>

#### `load()`

<p>

**Type**: `(id: string) => Promise<void>`

> 
> <Since v="5.6.0" />
> 

Reviewer: ```suggestion
```
Removing, since the entire entry will be stable in 5.**7**.0

</p>

> Loads a session by ID. In normal use a session is loaded automatically from the request cookie, but this method can be used to load a session from a different ID. This is useful if you are handling the session ID yourself, or if you want to keep track of a session without using cookies.

Reviewer: ```suggestion
Loads a session by ID. In normal use, a session is loaded automatically from the request cookie. Use this method to load a session from a different ID. This is useful if you are handling the session ID yourself, or if you want to keep track of a session without using cookies.
```
